tests/integration/test_db.py

```diff
@@ -156,3 +158,39 @@ async def test_relation_data_is_updated_correctly_when_scaling(ops_test: OpsTest
         await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=1000)
         with pytest.raises(psycopg2.OperationalError):
             psycopg2.connect(primary_connection_string)
+
+
+@pytest.mark.db_relation_tests
+async def test_nextcloud_db_blocked(ops_test: OpsTest, charm: str) -> None:
+    async with ops_test.fast_forward():
+        # Deploy Nextcloud.
+        await ops_test.model.deploy(
+            "nextcloud",
+            channel="edge",
+            application_name="nextcloud",
+            num_units=APPLICATION_UNITS,
+        )
+        await ops_test.model.wait_for_idle(
+            apps=["nextcloud"],
+            status="blocked",
+            raise_on_blocked=False,
+            timeout=1000,
+        )
```

Comment:
This is a third party charm requesting extensions.

Reply:
Awesome finding!